polaris-react/src/components/List/List.scss

```diff
@@ -25,3 +23,9 @@
     margin-top: var(--p-space-2);
   }
 }
+
+.spacingLoose {
```

Comment:
This probably needs to reset `margin-bottom` to `0` on the last child as well. It's what's causing the [unintended UI changes](https://shopify.chromatic.com/build?appId=5d559397bae39100201eedc1&number=14920) for related components with extra padding on the last list item.

Reply:
Oh, I forgot polaris has chromatics, thank you for point it out!

Reply:
After the last commit it's looking way better. b531b257c8cb8f0575015c19b79cf2193fa87f47